File: Presentation/Models/HttpModels/Responses/PagedResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HttpModels.Responses
{
    public class PagedResponse<T>
    {
        public IEnumerable<T> Items { get; set; }
        public int TotalItems { get; set; }
    }
}

File: Presentation/cosmeticshop.client/src/apiClient/models/paged-response.ts
/**
 * 
 *
 * @export
 * @interface PagedResponse
 * @template T
 */
export interface PagedResponse<T> {

    /**
     * Список элементов типа T
     * @type {Array<T>}
     * @memberof PagedResponse
     */
    items: Array<T>;

    /**
     * Общее количество элементов
     * @type {number}
     * @memberof PagedResponse
     */
    totalItems: number;
}